airflow/models.py

```diff
@@ -962,11 +962,11 @@ class derived from this one results in the creation of a task object,
         sequentially while relying on the previous task's schedule to
         succeed. The task instance for the start_date is allowed to run.
     :type depends_on_past: bool
-    :param wait_for_downstream: when set to true, the task instances
-        of task X will wait for the dependencies of the previous instance
-        of task X to finish before it moves on the to next schedule.
-        This is useful if the different instances of a task X alter
-        the same asset, and this asset is used by the dependencies of task X.
+    :param wait_for_downstream: when set to true, an instance of task
+        X will wait for the dependents of the previous instance of task X
+        to finish successfully before it runs. This is useful if the
+        different instances of a task X alter the same asset, and this asset
+        is used by the dependencies of task X.
```

Comment:
dependencies->dependents on this line too

I try to avoid using the words dependents and dependencies in the docs and favor upstream and downstream instead.